src/bp-activity/actions/dislike.php

```diff
@@ -0,0 +1,69 @@
+<?php
+/**
+ * Activity: Dislike action
+ *
+ * @package BuddyPress
+ * @subpackage ActivityActions
+ * @since 14.0.0
+ */
+
+// Exit if accessed directly.
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * Dislike an activity.
+ *
+ * @since 14.0.0
+ *
+ * @return bool False on failure.
```

Comment:
This has a mixture of `bool` with `void`. Tools like `phpstan` would rightly complain it is missing a return value. I think you should bail without a `bool`.